examples/speech_recognition/modules/specaugment.py

```diff
@@ -0,0 +1,79 @@
+"""SpecAugment Implementation for Pytorch.
+Related paper : https://arxiv.org/pdf/1904.08779.pdf
+"""
+
+import torch
+import torch.nn as nn
+import numpy as np
+import random
+
+
+class SpecAugment(nn.Module):
+
+    def __init__(self, frequency_masking_pars, time_masking_pars,
+                 frequency_masking_num, time_masking_num, rate=1.0):
+        super().__init__()
+        self.frequency_masking_pars = frequency_masking_pars
+        self.time_masking_pars = time_masking_pars
+        self.frequency_masking_num = frequency_masking_num
+        self.time_masking_num = time_masking_num
+        self.rate = rate
+
+    def forward(self, batch):
+        new_spectrograms = []
+        x = batch['net_input']['src_tokens']
+        for spectrogram in x:
+            if random.random() < self.rate:
+                sample = specaugment(spectrogram, self.frequency_masking_pars,
+                                    self.time_masking_pars, self.frequency_masking_num, self.time_masking_num,
+                                     )
+            else:
+                sample = spectrogram
+            new_spectrograms += [sample]
+
+        new_spectrograms = torch.stack(new_spectrograms)
+        batch['net_input']['src_tokens'] = new_spectrograms
+        return batch
+
+
+def specaugment(mel_spectrogram, frequency_masking_para=27,
+                 time_masking_para=100, frequency_masking_num=1, time_masking_num=1):
+    """Spec augmentation Calculation Function.
+
+    'SpecAugment' have 3 steps for audio data augmentation.
+    first step is time warping using Tensorflow's image_sparse_warp function.
+    Second step is frequency masking, last step is time masking.
+
+    # Arguments:
+      mel_spectrogram(numpy array): audio file path of you want to warping and masking.
```

Comment:
it is not a path, but a numpy array containing the filterbanks, isn't it? Actually, my undestanding is that this is a tensor, not a numpy array, am I wrong?